src/main/java/org/jenkinsci/plugins/codesonar/services/HubVersionService.java

```diff
@@ -0,0 +1,151 @@
+             */
+            LOGGER.log(Level.WARNING, String.format("Falling back to legacy version endpoint (/command/anon_info/)"));
+            String hubVersionLegacy = fetchHubVersionLegacy(baseHubUri);
+            hubVersion.setVersion(hubVersionLegacy);
+        }
+
+        return hubVersion;
+    }
+    
+    /**
+     * 
+     * @param baseHubUri
+     * @param clientName the client name, in this case "jenkins"
+     * @param clientVersion the client version
+     * @throws IOException thrown when the communication with the hub fails
+     */
+    private VersionCompatibilityInfo fetchVersionCompatibilityInfo(URI baseHubUri, String clientName, String clientVersion) throws IOException {
+        
+        URI resolvedURI = baseHubUri;
+        
+        resolvedURI = baseHubUri.resolve(String.format("/command/check_version/%s/?version=%s&capability=openapi", clientName, clientVersion));
+        LOGGER.log(Level.WARNING, "Calling " + resolvedURI.toString());
+        
+        HttpResponse resp = httpService.execute(Request.Get(resolvedURI))
+                .returnResponse();
+        
+        Gson gson = new Gson();
+
+        HttpEntity entity = resp.getEntity();
+        if(entity == null) {
+            throw new AbortException("[CodeSonar] hub compatibility info cannot be read. %n[CodeSonar] entity is null");
+        }
+        
+        String responseBody = EntityUtils.toString(entity, Consts.UTF_8);
+        
+        //Hub might respond with just the text "Not Found", in this case we must fail
+        if(StringUtils.isNotEmpty(responseBody) && responseBody.equals("Not Found"))  {
+            throw new AbortException(String.format("[CodeSonar] hub compatibility info have not been returned. %n[CodeSonar] response is \"%s\"", responseBody));
+        }
+        
+        VersionCompatibilityInfo vci = gson.fromJson(responseBody, VersionCompatibilityInfo.class);
+        
+        //Returned object might be null if either "responseBody" is null or if it's empty, in such a case we must fail
+        if(vci == null) {
+            throw new AbortException(String.format("[CodeSonar] hub compatibility info cannot be parsed. %n[CodeSonar] response is \"%s\"", responseBody));
+        }
+        
+        LOGGER.log(Level.WARNING, vci.toString());
+        
+        return vci;
+    }
+    
+    private boolean supportsOpenAPI(VersionCompatibilityInfo vci) {
+        return vci.getCapabilities() != null
+                && vci.getCapabilities().getOpenapi() != null
+                && vci.getCapabilities().getOpenapi().booleanValue();
+    }
+
+    private boolean checkClientOk(VersionCompatibilityInfo vci) {
+        //If "clientOk" is either "true" or "null", this means that this client is supposed to be able to talk to the hub
+        return vci.getClientOK() == null || vci.getClientOK().booleanValue();
+    }
+    
+    private String fetchHubVersionLegacy(URI baseHubUri) throws AbortException {
```

Comment:
Maybe add a comment to this function to say where you copied it from, since I know you didn't write it new in this file.